app/src/main/java/org/greenstand/android/TreeTracker/login/LoginView.kt

```diff
@@ -0,0 +1,221 @@
+package org.greenstand.android.TreeTracker.login
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.text.KeyboardActionScope
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.SnackbarHostState
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.platform.LocalTextInputService
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import kotlinx.coroutines.launch
+import org.greenstand.android.TreeTracker.R
+import org.greenstand.android.TreeTracker.utilities.Padding
+import org.greenstand.android.TreeTracker.utilities.isValidEmail
+import org.greenstand.android.TreeTracker.utilities.isValidPhoneNumber
+import org.greenstand.android.TreeTracker.view.TreeTrackerButton
+import org.greenstand.android.TreeTracker.view.TreeTrackerTheme
+
+@Composable
+fun LoginView(
+    viewModel: LoginViewModel,
+    navController: NavController
+) {
+
+    val uiScope = rememberCoroutineScope()
+    val snackBarHostState = remember { SnackbarHostState() }
+    val phoneTextInputState = mutableStateOf("")
```

Comment:
I'm still new to compose, but shouldn't mutableStateOf() be in a remember block?

Reply:
Yes! Good catch!